src/main/java/lotto/controller/LottoController.java

```diff
@@ -0,0 +1,50 @@
+package lotto.controller;
+
+import lotto.model.lotto.Lotto;
+import lotto.model.user.UserLotto;
+import lotto.model.user.UserLottoRank;
+import lotto.model.user.UserMoney;
+import lotto.model.lotto.WinningNumbers;
+import lotto.util.generator.NumberGenerator;
+import lotto.view.UserIoManager;
+
+public class LottoController {
+    private final UserIoManager userIoManager;
+    private final NumberGenerator numberGenerator;
+
+    public LottoController(UserIoManager userIoManager, NumberGenerator numberGenerator) {
+        this.userIoManager = userIoManager;
+        this.numberGenerator = numberGenerator;
+    }
+
+    public void run() {
+        UserMoney userMoney = readMoney();
+
+        UserLotto userLotto = generateUserLotto(userMoney);
+
+        WinningNumbers winningNumbers = readWinningNumbers();
+
+        printResult(userMoney, userLotto, winningNumbers);
+    }
+
+    private UserMoney readMoney() {
+        return userIoManager.readUserMoney();
+    }
+
+    private UserLotto generateUserLotto(UserMoney userMoney) {
+        userIoManager.printLottoAmount(userMoney);
+        UserLotto userLotto = new UserLotto(userMoney.getLottoAmount(), numberGenerator);
+        userIoManager.printUserLotto(userLotto);
+        return userLotto;
+    }
+
+    private WinningNumbers readWinningNumbers() {
+        Lotto lotto = userIoManager.readWinningNumbers();
+        return userIoManager.readBonusNumber(lotto);
+    }
+
+    private void printResult(UserMoney userMoney, UserLotto userLotto, WinningNumbers winningNumbers) {
+        UserLottoRank userLottoRank = new UserLottoRank(userLotto, winningNumbers);
+        userIoManager.printResult(userMoney, userLottoRank);
+    }
```

Comment:
함수 이름만 보고 결과를 출력만 해줄 줄 알았는데, 실제로는 로또 등수도 계산하고 있습니다. 의도하신 걸까요?

Reply:
지금 생각해보니 등수를 생성하는 `new UserLottoRank()`를 분리했어야 했네요 😅
수정한다면 이렇게 분리할 것 같아요!

```suggestion
    private UserLottoRank generateUserLottoRank(UserLotto userLotto, WinningNumbers winningNumbers) {
        return new UserLottoRank(userLotto, winningNumbers);
    }

    private void printResult(UserMoney userMoney, UserLottoRank userLottoRank) {
        userIoManager.printResult(userMoney, userLottoRank);
    }
```